core/devmode-spi/src/main/java/io/quarkus/dev/testing/results/TestClassResultInterface.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.dev.testing.results;
+
+import java.util.List;
+
+public interface TestClassResultInterface extends Comparable<TestClassResultInterface> {
```

Comment:
Could you clarify the purpose of all these new interfaces?

Reply:
The test results are stored into objects of type `TestRunResults`, `TestClassResult` and `TestResult`. They are returned to the `ContinuousTestingJsonRPCService` as a generic `java.lang.Object` and there transformed to JSON - then sent to the browser for UI rendering. Because my intention was to avoid passing the objects through the RPC service without any API awareness, validation..., the RPC service needs the objects typified.
Unfortunately, the module, where the JSON RPC service is placed, does not have a compile dependency to the module where the Test result types are declared. (And I did not want to change this, because this might lead to a cyclic dependency.) So I declared the interfaces (whose names might sound uncreative, because I did not want to change the implementation classes' names to avoid too much diffs) in one module where both have access to. (I'm not sure if this is the correct module, but I found the `ContinuousTestingSharedStateManager` there, so I think it is intended so)

Reply:
Furthermore, the interfaces have the advantage that we can be aware of which informationen in required for the Continuous Testing UI, and which information is not. (that is not part of the interface) This makes further refactoring easier, because we now only have compile-time dependencies. (The test results are also used for printing out in the console for example.)

Reply:
What is unclear for me: Why are the classes `TestSupport`, `TestRunResults`, `TestClassResult` and `TestResult`placed into a deployment module (`core-deployment`)? They could also be placed into `development-mode-spi`, where the interfaces are now? This might avoid the need of the interfaces.